src/isomorphic/modern/class/ReactBaseClasses.js

```diff
@@ -21,7 +22,36 @@ var warning = require('fbjs/lib/warning');
 /**
  * Base class helpers for the updating state of a component.
  */
+
+const addUpdaterMutationWarn = (() => {
+  const isValidUpdater = value =>
+    value &&
+    (Object.keys(ReactNoopUpdateQueue).every(key =>
+      value.hasOwnProperty(key)) ||
+      value instanceof ReactServerUpdateQueue);
```

Comment:
A valid updater is any of the ones React uses internally. So this also includes the updaters used by [Stack](https://github.com/facebook/react/blob/master/src/renderers/shared/stack/reconciler/ReactUpdateQueue.js#L70) and [Fiber](https://github.com/facebook/react/blob/master/src/renderers/shared/fiber/ReactFiberClassComponent.js#L67). (Be aware that there's a module called `ReactFiberUpdateQueue` but that's not an updater. An unfortunate naming collision.)

Instead of validating the shape of the object, I would add a DEV-only property to each of our updaters and check for that.